masonry_core/src/properties/types/gradient.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright 2025 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::kurbo::{Point, Rect};
+use crate::peniko::color::{ColorSpaceTag, HueDirection};
+use crate::peniko::{ColorStops, ColorStopsSource, Extend};
+
+/// Properties for the supported [`Gradient`] types.
+///
+/// This mirrors [`peniko::GradientKind`](crate::peniko::GradientKind),
+/// but uses a layout-invariant representation: instead of saying
+/// "The gradient goes from point A to point B", we declare things like
+/// "The gradient has angle X", and A and B and computed dynamically from widget layout.
+#[derive(Clone, Debug)]
+pub enum GradientShape {
+    /// Gradient that transitions between two or more colors along a line.
+    ///
+    /// This is interpreted like [`linear-gradient()`] in CSS.
+    ///
+    /// [`linear-gradient()`]: https://drafts.csswg.org/css-images-3/#linear-gradient-syntax
+    Linear {
+        /// The angle defining the gradient line's direction, in radians.
+        /// Zero points upwards and positive angles represent clockwise rotation.
+        angle: f64,
+    },
+}
+
+/// Definition of a gradient that transitions between two or more colors.
+///
+/// This mirrors [`peniko::Gradient`](crate::peniko::Gradient),
+/// but uses a layout-invariant representation: instead of saying
+/// "The gradient goes from point A to point B", we declare things like
+/// "The gradient has angle X", and A and B and computed dynamically from widget layout.
+#[derive(Clone, Debug)]
+pub struct Gradient {
+    /// Shape and coordinates of the gradient.
+    pub shape: GradientShape,
+    /// Extend mode.
+    pub extend: Extend,
+    /// The color space to be used for interpolation.
+    ///
+    /// The colors in the color stops will be converted to this color space.
+    ///
+    /// This defaults to [sRGB](ColorSpaceTag::Srgb).
+    pub interpolation_cs: ColorSpaceTag,
+    /// When interpolating within a cylindrical color space, the direction for the hue.
+    ///
+    /// This is interpreted as described in [CSS Color Module Level 4 § 12.4].
+    ///
+    /// [CSS Color Module Level 4 § 12.4]: https://drafts.csswg.org/css-color/#hue-interpolation
+    pub hue_direction: HueDirection,
+    /// Color stop collection.
+    pub stops: ColorStops,
+}
+
+impl Gradient {
+    /// Creates a [`Linear`](GradientShape::Linear) gradient.
+    ///
+    /// `angle` is in radians, with zero pointing upwards.
```

Comment:
It might be worth adding something like:
```suggestion
    /// `angle` is in radians, with zero pointing upwards, and higher values rotating the gradient clockwise.
    /// This matches how [CSS gradients are defined](https://developer.mozilla.org/en-US/docs/Web/CSS/gradient/linear-gradient#angle).
    ///
    /// That is, for an `angle` of 0, the first stop will be at the bottom, and
    /// for an `angle` of [`π/2`](core::f32::consts::FRAC_PI_2) (90°), the first
    /// stop will be aligned with the left edge.
```

Since increasing radians almost always go anticlockwise usually (and 0 indicates pointing to the left)